WikipediaUnitTests/Code/ArticlePerformanceTests.swift

```diff
@@ -0,0 +1,122 @@
+import XCTest
+@testable import Wikipedia
+
+class MeasurableArticleViewController: ArticleViewController {
+    
+    var initialLoadCompletion: (() -> Void)?
+    
+    override func handlePCSDidFinishFinalSetup() {
+        super.handlePCSDidFinishFinalSetup()
+        initialLoadCompletion?()
+    }
+}
+
+class MeasurableArticlePeekPreviewViewController: ArticlePeekPreviewViewController {
+    var displayCompletion: (() -> Void)?
+    
+    override func updatePreferredContentSize(for contentWidth: CGFloat) {
+        super.updatePreferredContentSize(for: contentWidth)
+        displayCompletion?()
+    }
+}
+
+class ArticlePerformanceTests: XCTestCase {
+    
+    private var articleURL: URL! = URL(string: "https://en.wikipedia.org/wiki/Dog")
+    private var appSchemeArticleURL: URL! = URL(string: "app://en.wikipedia.org/wiki/Dog")
+    
+    override func setUp() {
+        super.setUp()
+        
+        LSNocilla.sharedInstance().start()
+        ArticleTestHelpers.stubCompleteMobileHTMLResponse(inBundle: wmf_bundle())
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        LSNocilla.sharedInstance().stop()
+    }
+
+    //represents the speed at which article content is seen on screen
+    func testArticleSetupTime() {
+
+        self.measure {
+            
+            let dataStore = MWKDataStore.temporary()
+            guard let measurableArticleVC = MeasurableArticleViewController(articleURL: articleURL, dataStore: dataStore, theme: .light) else {
+                XCTFail("Unable to instantiate MeasurableArticleViewController")
+                return
+            }
+            
+            let setupExpectation = expectation(description: "Waiting for article initial setup call")
+            
+            measurableArticleVC.initialLoadCompletion = {
+                setupExpectation.fulfill()
+                UIApplication.shared.keyWindow?.rootViewController = nil
+                dataStore.clearTemporaryCache()
+            }
+            
+            UIApplication.shared.keyWindow?.rootViewController = measurableArticleVC
+            let _ = measurableArticleVC.view
+        
+            wait(for: [setupExpectation], timeout: 10)
```

Comment:
This implies a 10 second wait, right? I know we talked a bit about how long this should wait for... but 10 seconds seems high, if that is what it is.

Reply:
Agreed, good catch. I've bumped them down to 3 seconds.